oci/private/image.sh

```diff
@@ -1,104 +1,88 @@
+trap "on_exit" EXIT
+
+# Redirect stderr to the $STDERR temp file for the rest of the script.
+exec 2>>"${STDERR}"
+
+# shellcheck disable=SC1090
+source "${REGISTRY_IMPL}" 
+REGISTRY=
+REGISTRY=$(start_registry "${OUTPUT}" "${REGISTRY_STDERR}")
 
 
-function empty_base() {
-    local registry=$1
-    local ref="$registry/oci/empty_base:latest"
-    ref="$("${CRANE}" append --oci-empty-base -t "${ref}" -f {{empty_tar}})"
+function base_from_scratch() {
+    local raw_platform="$1"
+    local ref="${REGISTRY}/scratch:latest"
+    ref="$("${CRANE}" append --oci-empty-base -t "${ref}" "-f" "{{empty_tar}}")"
     ref=$("${CRANE}" config "${ref}" | "${JQ}"  ".rootfs.diff_ids = [] | .history = []" | "${CRANE}" edit config "${ref}")
     ref=$("${CRANE}" manifest "${ref}" | "${JQ}"  ".layers = []" | "${CRANE}" edit manifest "${ref}")
 
-    local raw_platform=$(get_option --platform $@)
     IFS='/' read -r -a platform <<< "$raw_platform"
 
-    local filter='.os = $os | .architecture = $arch'
-    local -a args=( "--arg" "os" "${platform[0]}" "--arg" "arch" "${platform[1]}" )
+    local filter=".os = \"${platform[0]}\" | .architecture = \"${platform[1]}\""
 
     if [ -n "${platform[2]:-}" ]; then
-        filter+=' | .variant = $variant'
-        args+=("--arg" "variant" "${platform[2]}")
-    fi
-    "${CRANE}" config "${ref}" | "${JQ}" ${args[@]} "${filter}" | "${CRANE}" edit config "${ref}"
-}
-
-function base_from_layout() {
-    # TODO: https://github.com/google/go-containerregistry/issues/1514
-    local refs=$(mktemp)
-    local output=$(mktemp)
-    local oci_layout_path=$1
-    local registry=$2
-
-    "${CRANE}" push "${oci_layout_path}" "${registry}/image:latest" --image-refs "${refs}" > "${output}" 2>&1
-
-    if grep -q "MANIFEST_INVALID" "${output}"; then
-    cat >&2 << EOF
-
-zot registry does not support docker manifests. 
-
-crane registry does support both oci and docker images, but is more memory hungry.
-
-If you want to use the crane registry, remove "zot_version" from "oci_register_toolchains". 
-
-EOF
-
-        exit 1
+        filter+=" | .variant = \"${platform[2]}\""
     fi
-
-    cat "${refs}"
+    "${CRANE}" config "${ref}" | "${JQ}" "${filter}" | "${CRANE}" edit config "${ref}"
 }
 
-# Redirect stderr to the $STDERR temp file for the rest of the script.
-exec 2>>"${STDERR}"
-
-# Upon exiting, stop the registry and print STDERR on non-zero exit code.
-trap "on_exit" EXIT
-
-source "${REGISTRY_LAUNCHER}" 
-REGISTRY=
-REGISTRY=$(start_registry "${STORAGE_DIR}" "${STDERR}")
-
-OUTPUT=""
 FIXED_ARGS=()
 ENV_EXPANSIONS=()
 
 for ARG in "$@"; do
     case "$ARG" in
-        (oci:registry*) FIXED_ARGS+=("${ARG/oci:registry/$REGISTRY}") ;;
-        (oci:empty_base) FIXED_ARGS+=("$(empty_base $REGISTRY $@)") ;;
-        (oci:layout*) FIXED_ARGS+=("$(base_from_layout ${ARG/oci:layout\/} $REGISTRY)") ;;
-        (--output=*) OUTPUT="${ARG#--output=}" ;;
+        (--scratch=*)
+          # https://www.shellcheck.net/wiki/SC2155
```

Comment:
nit: SC2155 doesn't apply here since we're not using `export` or `local`

Reply:
yep, i found that `+=()` was masking the error.

```
# this fails
REF=$(item)
A+=($REF)

# this doesn't fail
A+=($(item))
```